narwhal/worker/src/transactions_server.rs

```diff
@@ -179,10 +183,18 @@ impl<V: TransactionValidator> Transactions for TxReceiverHandler<V> {
             }
             // Send the transaction to the batch maker.
             let (notifier, when_done) = tokio::sync::oneshot::channel();
-            self.tx_batch_maker
-                .send((txn.transaction.to_vec(), notifier))
-                .await
-                .expect("Failed to send transaction");
+            match self
+                .tx_batch_maker
+                .try_send((txn.transaction.to_vec(), notifier))
+            {
+                Ok(()) => (),
+                Err(TrySendError::Full(_)) => {
```

Comment:
I like this since it signals back to the client / validator to back off for a bit. However, should we also handle this on the validator side (maybe another PR).

Reply:
Is there any validator-side handling in particular you're looking for? I need to figure out retries (and if load shedding could be possible) in `ConsensusAdapter`, anything beyond that?